src/System.Numerics.Vectors/tests/GenericVectorTests.netcoreapp.cs

```diff
@@ -212,15 +284,121 @@ public partial class GenericVectorTests
+        [Fact]
+        public void CopyToSpanUInt32() { TestCopyToSpan<uint>(); }
+        [Fact]
+        public void CopyToSpanInt32() { TestCopyToSpan<int>(); }
+        [Fact]
+        public void CopyToSpanUInt64() { TestCopyToSpan<ulong>(); }
+        [Fact]
+        public void CopyToSpanInt64() { TestCopyToSpan<long>(); }
+        [Fact]
+        public void CopyToSpanSingle() { TestCopyToSpan<float>(); }
+        [Fact]
+        public void CopyToSpanDouble() { TestCopyToSpan<double>(); }
+        private void TestCopyToSpan<T>() where T : struct
+        {
+            var initialValues = GenerateRandomValuesForVector<T>();
+            var vector = new Vector<T>(initialValues);
+            Span<T> destination = new T[Vector<T>.Count];
+
+            Assert.Throws<ArgumentException>(() => vector.CopyTo(new Span<T>(new T[Vector<T>.Count - 1])));
+
+            // CopyTo(Span<T>) method
+            vector.CopyTo(destination);
+            for (int g = 0; g < destination.Length; g++)
+            {
+                Assert.Equal(initialValues[g], destination[g]);
+                Assert.Equal(vector[g], destination[g]);
+            }
+
+            destination.Clear();
+
+            Assert.Throws<ArgumentException>(() => vector.CopyTo(new Span<byte>(new byte[Vector<byte>.Count - 1])));
+
+            // CopyTo(Span<byte>) method
+            vector.CopyTo(MemoryMarshal.AsBytes(destination));
+            for (int g = 0; g < destination.Length; g++)
+            {
+                Assert.Equal(initialValues[g], destination[g]);
+                Assert.Equal(vector[g], destination[g]);
+            }
+
+        }
+        #endregion
+
+        #region TryCopyTo (span) Tests
+        [Fact]
+        public void TryCopyToSpanByte() { TestTryCopyToSpan<byte>(); }
+        [Fact]
+        public void TryCopyToSpanSByte() { TestTryCopyToSpan<sbyte>(); }
+        [Fact]
+        public void TryCopyToSpanUInt16() { TestTryCopyToSpan<ushort>(); }
+        [Fact]
+        public void TryCopyToSpanInt16() { TestTryCopyToSpan<short>(); }
+        [Fact]
+        public void TryCopyToSpanUInt32() { TestTryCopyToSpan<uint>(); }
+        [Fact]
+        public void TryCopyToSpanInt32() { TestTryCopyToSpan<int>(); }
+        [Fact]
+        public void TryCopyToSpanUInt64() { TestTryCopyToSpan<ulong>(); }
+        [Fact]
+        public void TryCopyToSpanInt64() { TestTryCopyToSpan<long>(); }
+        [Fact]
+        public void TryCopyToSpanSingle() { TestTryCopyToSpan<float>(); }
+        [Fact]
+        public void TryCopyToSpanDouble() { TestTryCopyToSpan<double>(); }
+        private void TestTryCopyToSpan<T>() where T : struct
+        {
+            var initialValues = GenerateRandomValuesForVector<T>();
+            var vector = new Vector<T>(initialValues);
+            Span<T> destination = new T[Vector<T>.Count];
+
+            Assert.False(vector.TryCopyTo(destination.Slice(1)));
```

Comment:
Do we want to validate anything about the contents of destination after the copy fails?  I'm ok if the answer is "it may or may not have written data", just checking.